Documentation/design-docs/targeting-packs-on-nuget.md

```diff
@@ -0,0 +1,314 @@
+- Providing a way to allow for supporting new targets with PCL profiles and not break everyone's code
+
+A long-term goal is to make this the mechanism Visual Studio uses, as well.
+
+## Let's talk solutions
+
+1. Author NuGet Packages adhering to a particular naming scheme:
+
+   - `Microsoft.TargetingPack.NETFramework.v3.5`
+   - `Microsoft.TargetingPack.NETFramework.v4.6.1`
+   - `Microsoft.TargetingPack.NETPortable.Profile259`
+   - etc.
+   
+2. Implement a convention where the CLI treats the NuGet package as the replacement directory for framework assemblies.
+
+   - Instead of looking in e.g., `ref/net46`, it looks in the package directory instead.
+   - Will prefer targeting packs already installed on the machine if they exist (so as not to screw with VS).
+  
+3. Have the targeting pack define metadata that informs the build system.
+
+   - This will solve the problem of traditional PCL targets not auto-referencing their reference assemblies.
+   - PCL packs will specify what the NuGet TFM for that particular profile is.
+   - Allowing targeting packs to specify this info makes the .NET CLI happy.
+
+4. Use this as an avenue for getting rid of `frameworkAssemblies`, instead allowing you to reference directly from the package.
+
+   - This requires NuGet and the existing project system to allow referencing directly from within a package.
+   - Reasoning about "additionally pulling in these references from this package" makes the most sense.
+
+5. For now, only worry about English.
+
+   - We don't have a good localization story for .NET Core yet, but it may be important if this is how we supply the data for intellisense in the future.
+   
+6. For RTM, keep this confined to the CLI.
+
+   - Let's not break Intellisense in VS2015.
+
+## So what gets a targeting pack?
+
+Each version of .NET Framework gets its own targeting pack:
+
+- 2.0
+- 3.5
+- 4.0
+- 4.5
+- 4.5.1
+- 4.5.2
+- 4.6
+- 4.6.1
+- future versions of .NET Framework
+
+Each PCL profile gets its own targeting pack.
+
+Silverlight 5 gets a targeting pack.
+
+Xamarin targeting packs may be created in the future.
+
+## Show me the ~~money~~ JSON
+
+Here's what the targeting section of JSON.NET's `project.json` file could look like with this implemented.
+
+```javascript
+{
+   ...
+    "frameworks": {
+        "net45": {
+            "dependencies":{
+               "Microsoft.TargetingPack.NETFramework.v4.5":{
+                  "version":"1.0.0",
+                  "references":[
+                     "System.Data",
+                     "System.Numerics",
+                     "System.Runtime.Serialization",
+                     "System.Xml",
+                     "System.Xml.Linq"
+                  ]
+               }
+            }
+       },
+       "net40": {
+            "dependencies":{
+               "Microsoft.TargetingPack.NETFramework.v4.0":{
+                  "version": "1.0.0",
+                  "references": [
```

Comment:
Why do we need a `references` section for the desktop frameworks? Is there harm in passing all the libraries in? One current issue is that a few of the files in the targeting pack are not .NET assemblies, but I would think putting those in a `ref` directory would solve that?

Main point is to not make the user do all the work and to treat the targeting pack just like the `netstandard.library` where it pulls in the world.